src/app/integrations/edit-page/step-configure/filter-steps/basic-filter.model.ts

```diff
@@ -108,10 +108,13 @@ export function createBasicFilterModel(configuredProperties: BasicFilter) {
     ];
   };
   let groups = undefined;
+  let rules = undefined;
   // build up the form array from the incoming values (if any)
   if (configuredProperties && configuredProperties.rules && configuredProperties.rules.length) {
+    // TODO hackity hack
+    rules = JSON.parse(<any>configuredProperties.rules);
```

Comment:
Not sure whether this is correct in this place, has configuredProperties is of type `BasicFilter` which in turn contains the rules as an array of `Rule` elements already.